Plot customization example (#228)
PhiLFitters/kafe2: doc/src/parts/user_guide.rst

Customize the Plot
------------------

Model and parameter labels
^^^^^^^^^^^^^^^^^^^^^^^^^^

The model and parameter labels that appear in the plot info box can be set like this:
.. code-block:: python
    # Assign the LaTeX-names that are used in the plot info box (legend):

>     fit.assign_parameter_latex_names(x='t', a='\\alpha', b='\\beta')
>     fit.assign_model_function_latex_name('\\theta')
>     fit.assign_model_function_latex_expression('{a} \\cdot {x} + {b}')

Reviewer: ```suggestion
    fit.assign_parameter_latex_names(x='t', a=r'\alpha', b=r'\beta')
    fit.assign_model_function_latex_name(r'\theta')
    fit.assign_model_function_latex_expression(r'{a} \cdot {x} + {b}')
```

Also it should be mentioned somewhere that when parameters are called alpha_1, beta, gamma_d, etc. in code they are automatically converted to LaTeX.

    plot.customize('model_error_band', 'color', '#DB1F0B')  # Set the color of the error band
    plot.customize('model_error_band', 'label', 'model error band label')  # Set the label for the error band

    # Analogous to data: set the appearance for the residual
    plot.customize('residual', 'marker', 'X')
    plot.customize('residual', 'markersize', 10)
    plot.customize('residual', 'color', '#600E8F')
    plot.customize('residual', 'ecolor', '#8F0BDB')

    # Analogous to model error band: set the appearance for the residual error band
    plot.customize('residual_error_band', 'alpha', 0.2)
    plot.customize('residual_error_band', 'linestyle', '--')
    plot.customize('residual_error_band', 'linewidth', 3)
    plot.customize('residual_error_band', 'color', '#DB1F0B')

To hide entries in the info box you can do this:
.. code-block::python
    plot.customize('data', 'label', None)

To hide specific elements of the plot (e.g. error bands) you can do this:
.. code-block::python
    plot.customize('model_error_band', 'hide', True)

Additionally it is possible to change parameters using matplotlib functions.

> Changing the size of the axis labels is done with the following calls:

Reviewer: ```suggestion
For example, the size of the axis labels can be changed with the following calls:
```

.. code-block:: python
    import matplotlib as mpl
    mpl.rc('axes', labelsize=20, titlesize=25)

Axis Range
^^^^^^^^^^

The plot range can be set via the :py:attr:`~.Plot.x_range` and :py:attr:`~.Plot.y_range` properties:
.. code-block:: python
    plot.x_range = (0.5, 4.5)  # Set the range of the x-axis
    plot.y_range = (2, 10)  # Set the range of the y-axis

Axis Labels
^^^^^^^^^^^

By default, the plot will use the labels specified for each dataset (see :ref:'container-labels').
If multiple fits are plotted to the same figure, the axis labels from the data containers are concatenated while skipping duplicates.

> Alternatively the axis labels can be overwritten for each fit.

Reviewer: I think this should be either "overridden for each fit" or "set as properties for each fit" depending on what you mean.

Again if multiple fits are plotted to the same figure, all labels will be concatenated while skipping duplicates.
.. code-block::python
    plot.x_label = 'x label'  # Overwrite the label of the x-axis
    plot.y_label = 'y label'  # Overwrite the label of the y-axis

Axis Scale
^^^^^^^^^^

> Additionally the axis scale can be changed to logarithmic.

Reviewer: ```suggestion
Additionally the axis scale can be set as logarithmic (base 10) rather than linear.
```

> When changing between a linear and logarithmic x-axis scale, the supporting points for plotting the model function will be updated and evenly spaced on a linear or logarithmic scale.

Reviewer: I don't know what this sentence is supposed to mean.

Reviewer 2: It's correct. The x values for drawing in the models will be spaced logarithmically in this case. But I think there's no need to mention this in the documentation as this is expected behaviour.

Reviewer 2: I'd suggest just to remove this sentence.
```suggestion

```

.. code-block::python
    plot.x_scale = 'linear'  # Set the scale of the x-axis to linear (default)
    plot.y_scale = 'log'  # Set the scale of the y-axis to logarithmic

The resulting plot will look like this:
.. figure::doc/src/_static/img/005_customize_customized_plot.png
    :alt: The customized plot

> If you want to plot multiple plot_objects you have to pass a list of matching length as the attribute:

Reviewer: ```suggestion
When plotting multiple fits in a single `Plot` object you need to pass a list to set properties on a per-fit basis.
```

.. code-block:: python

>     # if all plots will have the same attribute you can waive the list an d just pass the attribute

Reviewer: Please add an explicit example for this.

    plot.customize('model_error_band', 'hide', [True, False])

In order to change the name for the data set and suppress the second output, use the following call:
.. code-block:: python

>     plot.customize('data', 'label', [(0, "test data"), (1, '__del__')])

Reviewer: This syntax with `[(fit_index, property)]` was not introduced I think. Please document it.
